crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -0,0 +1,365 @@
+use super::{Argument, CallArguments, InferContext, Signature, Type};
+use crate::db::Db;
+use crate::types::diagnostic::{
+    INVALID_ARGUMENT_TYPE, MISSING_ARGUMENT, PARAMETER_ALREADY_ASSIGNED,
+    TOO_MANY_POSITIONAL_ARGUMENTS, UNKNOWN_ARGUMENT,
+};
+use crate::types::signatures::Parameter;
+use crate::types::UnionType;
+use ruff_python_ast as ast;
+
+/// Bind a [`CallArguments`] against a callable [`Signature`].
+///
+/// The returned [`CallBinding`] provides the return type of the call, the bound types for all
+/// parameters, and any errors resulting from binding the call.
+pub(crate) fn bind_call<'db>(
+    db: &'db dyn Db,
+    arguments: &CallArguments<'db>,
+    signature: &Signature<'db>,
+    callable_ty: Option<Type<'db>>,
+) -> CallBinding<'db> {
+    let param_count = signature.parameter_count();
+    let mut parameter_tys = vec![None; param_count];
```

Comment:
Should this be named `argument_tys` considering that we push the argument (and not parameter) types?

Reply:
I don't think so, because the most important fact about it is that each index in this vector corresponds to a parameter, in the order the parameters exist in the signature. There may be fewer arguments, in a different order (if keyword arguments are used). So these are the types assigned to each parameter, at this call site.

Reply:
Added a comment to clarify this.